src/Http/Routing/src/Builder/MapActionEndpointRouteBuilderExtensions.cs

```diff
@@ -76,5 +83,203 @@ public static class MapActionEndpointRouteBuilderExtensions
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static MapActionEndpointConventionBuilder MapPost(
+            this IEndpointRouteBuilder endpoints,
+            string pattern,
+            Delegate action)
+        {
+            return MapMethods(endpoints, pattern, PostVerb, action);
+        }
+
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches HTTP PUT requests
+        /// for the specified pattern.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="action">The delegate executed when the endpoint is matched.</param>
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that canaction be used to further customize the endpoint.</returns>
+        public static MapActionEndpointConventionBuilder MapPut(
+            this IEndpointRouteBuilder endpoints,
+            string pattern,
+            Delegate action)
+        {
+            return MapMethods(endpoints, pattern, PutVerb, action);
+        }
+
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches HTTP DELETE requests
+        /// for the specified pattern.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="action">The delegate executed when the endpoint is matched.</param>
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static MapActionEndpointConventionBuilder MapDelete(
+            this IEndpointRouteBuilder endpoints,
+            string pattern,
+            Delegate action)
+        {
+            return MapMethods(endpoints, pattern, DeleteVerb, action);
+        }
+
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches HTTP requests
+        /// for the specified HTTP methods and pattern.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="action">The delegate executed when the endpoint is matched.</param>
+        /// <param name="httpMethods">HTTP methods that the endpoint will match.</param>
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static MapActionEndpointConventionBuilder MapMethods(
```

Comment:
Do we need this? As a user, there are alternate ways to re-use the delegate if we need to. I'm also similarly against allowing more than one HTTP attributes per MapAction delegate.

Reply:
I agree that the flexibility of delegates means that we shouldn't need to make routing metadata overly complicated. That's why I removed `IRotePatternMetadata` and `IRouteOrderMetadata` in #30563. I also stopped implementing `IHttpMethodMetadata` in attributes which caused multiple instances to show up on endpoints that didn't have multiple instances before. https://github.com/dotnet/aspnetcore/pull/29878#discussion_r577913632

All that said, this is an overload that already exists for `RequestDelegate` and any argument about reusing the `Delegate`, you could make for reusing the `RequestDelegate`. I'm just keeping it consistent by adding a `Delegate` overload for every `RequestDelegate` overload.